### DISCUSSION TITLE
Dockers

### DISCUSSION THREAD

**Discussion by euroblaze** [OCA/odoo-community.org#155] [Polls]
This is the discussion for those who use Dockers (in production) or have used them in the past.

We have used the official [Odoo Docker, mainly for v15 instances](https://hub.docker.com/_/odoo). 
While we learnt a lot about running Odoo in the Docker, and intend to continue do so, we also realised a good number of important problems with the above.

These are:

1. The **filestore** is on the Docker!

2. We are unable to freely use the Odoo shell (**CLI**) to install, uninstall and upgrade modules (as far as I understood, because of communication between the application and database dockers).

3. The **logfile** is on the Docker! For troubleshooting, the technician has to inevitably enter the Docker's shell.

4. Rebuilding the Docker image using erases the database and filestore. This makes it impossible to load the latest Odoo (base) updates, security fixes etc.

In our research we found the following Dockers used in the community, in addition to the official Odoo ones, and maybe there are others..

- [camptocamp / docker-odoo-project](https://github.com/camptocamp/docker-odoo-project)
- [Elico-Corp / odoo-docker](https://github.com/Elico-Corp/odoo-docker)
- [falconsoft3d / docker-odoo-16](https://github.com/falconsoft3d/docker-odoo-16)
- [Tecnativa / doodba](https://github.com/Tecnativa/doodba)
- [minhng92 / odoo-16-docker-compose](https://github.com/minhng92/odoo-16-docker-compose)

What has everyone's experience been?
How did you resolve the above 4 problems? If they are documented, I would welcome links.

Thanks for your insights, and I'm leaving this question with a poll below.

**Comment by ap-wtioit** [comment]
Regarding your questions and how we solved them using doodba:
1) You can map the filestore to your docker host (if you really want this, e.g. to lessen the burden of a knowledge gap regarding where docker stores it's files, also have a look at /var/lib/docker/volumes/)
2) You can use the odoo shell with doodba
3) We are using graylog for our log files (see also https://github.com/odoo/odoo/pull/26965)

**Comment by theangryangel** [comment]
> The filestore is on the Docker!

Docker volumes if what you're looking for here.

> We are unable to freely use the Odoo shell (CLI) to install, uninstall and upgrade modules (as far as I understood, because of communication between the application and database dockers).

Exec the shell inside the container, or spin up a temporary container. This should not be a problem.

> The logfile is on the Docker! For troubleshooting, the technician has to inevitably enter the Docker's shell.

Depending on if this is development or production, either docker volume, log to stdout and use logging solution of your choice to collect from docker directly.

**Comment by kimsonvu** [comment]
It is same 4 points that you write with v17?
